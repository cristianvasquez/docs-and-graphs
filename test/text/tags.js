import { expect } from 'expect'

import toMatchSnapshot from 'expect-mocha-snapshot'
import {
  extractTags, removeTags,
} from '../../src/text/tags.js'

const tweets = [
  'An #autumn scene showing a beautiful #horse coming to visit me.',
  'My new favourite eatery in #liverpool and I mean superb! #TheBrunchClub #breakfast #food',
  '#nowplaying Pointer Sisters - Dare Me | #80s #disco #funk #radio']


expect.extend({ toMatchSnapshot })

describe('extractTags', async function () {
  for (const current of tweets) {
    it(current, async function () {
      const tags = extractTags(current)
      expect(tags).toMatchSnapshot(this)
    })
  }
})

describe('removeTags', async function () {
  for (const current of tweets) {
    it(current, async function () {

      const txt = removeTags(current)
      expect(txt).toMatchSnapshot(this)
    })
  }
})
