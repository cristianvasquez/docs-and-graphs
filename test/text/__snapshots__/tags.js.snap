// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractTags #nowplaying Pointer Sisters - Dare Me | #80s #disco #funk #radio 1`] = `
Array [
  "#nowplaying",
  "#disco",
  "#funk",
  "#radio",
]
`;

exports[`extractTags An #autumn scene showing a beautiful #horse coming to visit me. 1`] = `
Array [
  "#autumn",
  "#horse",
]
`;

exports[`extractTags My new favourite eatery in #liverpool and I mean superb! #TheBrunchClub #breakfast #food 1`] = `
Array [
  "#liverpool",
  "#TheBrunchClub",
  "#breakfast",
  "#food",
]
`;

exports[`removeTags #nowplaying Pointer Sisters - Dare Me | #80s #disco #funk #radio 1`] = `" Pointer Sisters - Dare Me | #80s   "`;

exports[`removeTags An #autumn scene showing a beautiful #horse coming to visit me. 1`] = `"An  scene showing a beautiful  coming to visit me."`;

exports[`removeTags My new favourite eatery in #liverpool and I mean superb! #TheBrunchClub #breakfast #food 1`] = `"My new favourite eatery in  and I mean superb!   "`;
